### PR TITLE
[tests-only][full-ci] adding test for creating link space share via permission endpoint

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
This PR add test scenario for creating space share via link using permission endpoint. Scenerio covered:
```feature
Scenario Outline:try to create a link share of a Personal and Shares drives using permissions endpoint
Scenario Outline::create a link share of a project-space drive using permissions endpoint
Scenario Outline:try to create an internal link share of a Personal and Shares drives using permissions endpoint
Scenario Outline:try to create an internal link share with password of a Personal and Shares drives using permissions endpoint
Scenario:create an internal link share of a project-space using permissions endpoint
Scenario: try to create an internal link share of a project-space with password using permissions endpoint
Scenario Outline:create a link share of a project-space with display name and expiry date using permissions endpoint
Scenario Outline: try to create a link share of a project-space with a password that is listed in the Banned-Password-List using permissions endpoint
Scenario Outline: create a link share of a project-space without password using permissions endpoint

```

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/9046

## How Has This Been Tested?

- test environment:
-  locally
- CI
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 1]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
...
+        }
+      }
+      """
+    Examples:
+      | permissions-role | drive    | message                                   |
+      | view             | Shares   | no share permission                       |
+      | edit             | Shares   | no share permission                       |
+      | upload           | Shares   | no share permission                       |
+      | createOnly       | Shares   | no share permission                       |
+#      | blocksDownload   | Shares   | invalid link type                         |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2349]:
```suggestion
      | blocksDownload   | Shares   | invalid link type                         |
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 2]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
       """
     And the public should be able to download file "textfile.txt" from the last link share with password "%public%" and the content should be "to share"
     And the public download of file "textfile.txt" from the last link share with password "$heLlo*1234*" should fail with HTTP status code "401" using shareNg
+
+  @issue-7879
+  Scenario Outline: try to create a link share of a Personal and Share drive using permission endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2305]:
```suggestion
  Scenario Outline: try to create a link share of a Personal and Shares drives using permission endpoint
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 3]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
...
+      | createOnly       | Shares   | no share permission                       |
+#      | blocksDownload   | Shares   | invalid link type                         |
+      | view             | Personal | cannot create link on personal space root |
+      | edit             | Personal | cannot create link on personal space root |
+      | upload           | Personal | cannot create link on personal space root |
+      | createOnly       | Personal | cannot create link on personal space root |
+#      | blocksDownload   | Personal | invalid link type                         |
+
+  @issue-7879
+  Scenario Outline: create a link share of a project-space drive with different permissions using permission endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2357]:
```suggestion
  Scenario Outline: create a link share of a project-space drive using permissions endpoint
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 4]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
...
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+      | upload           |
+      | createOnly       |
+#      | blocksDownload   |
+
+
+  Scenario Outline: try to create an internal link share of a Personal and Share drives using permission endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2423]:
```suggestion
  Scenario Outline: try to create an internal link share of a Personal and Shares drives using permissions endpoint
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 5]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
...
+        }
+      }
+      """
+    Examples:
+      | drive    |
+      | Personal |
+      | Shares   |
+
+
+  Scenario: create an internal link share of a project-space using permission endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2510]:
```suggestion
  Scenario: create an internal link share of a project-space using permissions endpoint
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 6]
```diff
@@ -2300,3 +2300,374 @@ Feature: Create a share link for a resource
...
+                "const": "password is redundant for the internal link"
+              }
+            }
+          }
+        }
+      }
+      """
+
+
+  Scenario Outline: create a link share of a project-space drive with different permissions using permission endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2610]: dubplicate scenario?

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature (hunk 1 above)
Comment (S-Panta) [reply tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2349]: currently block download doesn't work

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2349]: yes, but we don't comment out the scenario. we put them in expected-to fail list

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 7]
```diff
@@ -2300,3 +2300,504 @@ Feature: Create a share link for a resource
       """
     And the public should be able to download file "textfile.txt" from the last link share with password "%public%" and the content should be "to share"
     And the public download of file "textfile.txt" from the last link share with password "$heLlo*1234*" should fail with HTTP status code "401" using shareNg
+
+  @issue-7879
+  Scenario Outline: try to create a link share of a Personal and Share drives using permissions endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2305]:
```suggestion
  Scenario Outline: try to create a link share of a Personal and Shares drives using permissions endpoint
```

**Review by saw-jan — APPROVED** [review]

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 8]
```diff
@@ -2300,3 +2300,504 @@ Feature: Create a share link for a resource
...
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+      | upload           |
+      | createOnly       |
+      | blocksDownload   |
+
+
+  Scenario Outline: try to create an internal link share of a Personal and Share drives using permissions endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2423]:
```suggestion
  Scenario Outline: try to create an internal link share of a Personal and Shares drives using permissions endpoint
```

File: tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature [hunk 9]
```diff
@@ -2300,3 +2300,504 @@ Feature: Create a share link for a resource
...
+        }
+      }
+      """
+    Examples:
+      | drive    | message                                   |
+      | Personal | cannot create link on personal space root |
+      | Shares   | no share permission                       |
+
+
+  Scenario Outline: try to create an internal link share with password of a Personal and Share drives using permissions endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature:2466]:
```suggestion
  Scenario Outline: try to create an internal link share with password of a Personal and Shares drives using permissions endpoint
```

**Pull request merged**